dashboard/app/reporting_external.go

```diff
@@ -85,3 +85,16 @@ func apiReportingUpdate(c context.Context, r *http.Request, payload []byte) (int
 		Text:  reason,
 	}, nil
 }
+
+func apiNewTestJob(c context.Context, r *http.Request, payload []byte) (interface{}, error) {
+	req := new(dashapi.TestPatchRequest)
+	if err := json.Unmarshal(payload, req); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal request: %v", err)
+	}
+	resp := &dashapi.TestPatchReply{}
+	err := handleExternalTestRequest(c, req)
+	if err != nil {
```

Comment:
As far as I see most of the errors here are actually our logic bugs and are not supposed to happen. We need to log them as errors to see on the admin page.

Reply:
Modified the code to log non-BadTestRequestErrors